lib/libc/string/strstr.c

```diff
@@ -149,8 +149,12 @@ twoway_strstr(const unsigned char *h, const unsigned char *n)
 	for (;;) {
 		/* Update incremental end-of-haystack pointer */
 		if (z - h < l) {
-			/* Fast estimate for MIN(l,63) */
+#ifdef __CHERI_PURE_CAPABILITY__
+			size_t grow = l;
+#else
+			/* Deliberate UB to avoid byte-wise path in memchr */
 			size_t grow = l | 63;
```

Comment:
I mean, it seems particularly poor given how unaligned this is going to be on subsequent iterations (and even the first time round you have a trailing byte-by-byte copy loop)... I assume it's `| 63` to avoid crossing page boundaries, but who says the initial `z` is 64-byte aligned? Maybe the maths makes it all work out but I am not convinced.

Reply:
`z` definitely isn't 64-bit aligned.  Here's a real example:
```
Breakpoint 2, memchr (src=0x7ffffefd86 [rwRW,0x7ffffefd80-0x7ffffefd8d], 
    src@entry=0x7ffffefd86 [xRW,0xfffffee000000000-0xffffffffffffffff] (sealed), c=c@entry=0, n=63, n@entry=6)
```
This whole thing should probably just be removed.